### PR TITLE
PODS-6854 Add schema validation code

### PR DESCRIPTION
**Pull request by grantrobson**

### CODE REVIEW

File: app/utils/JSONPayloadSchemaValidator.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+object ErrorDetailsExtractor {
+  val schemaPath = "/resources/schemas/api-1538-file-aft-return-request-schema-0.1.0.json"
+  val key = "/oneOf/0"
+
+  def getErrors(error: Seq[(JsPath, Seq[JsonValidationError])]): String = {
+    implicit val readSchemaErrorPayload = Json.reads[SchemaErrorPayload]
+    implicit val readExtractErrorDetails = Json.reads[ErrorDetailsExtractor]
+
+    val message = new StringBuilder("")
+    error.flatMap(x => x._2).foldLeft(message){
```
Comment (grantrobson) [inline app/utils/JSONPayloadSchemaValidator.scala:40]: x => x can be changed to _

File: app/utils/JSONPayloadSchemaValidator.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      (a, b) =>
```
Comment (grantrobson) [inline app/utils/JSONPayloadSchemaValidator.scala:41]: Can we use meaningful names here?

File: app/controllers/AFTController.scala [hunk 3]
```diff
@@ -61,19 +64,23 @@ class AFTController @Inject()(
...
             case JsSuccess(dataToBeSendToETMP, _) =>
-              logger.debug(message = s"[Compile File Return: Outgoing-Payload]$dataToBeSendToETMP")
-              aftConnector.fileAFTReturn(pstr, journeyType.toString, dataToBeSendToETMP).map { response =>
-                Ok(response.body)
+              val validationResult = jsonPayloadSchemaValidator.validateJsonPayload(s"$basePath/conf/${ErrorDetailsExtractor.schemaPath}",dataToBeSendToETMP)
+              validationResult match {
+                case JsError(error) =>
+                  val errors = ErrorDetailsExtractor.getErrors(error)
+                  throw AFTValidationFailureException(s"Invalid AFT file AFT return:-\n$errors")
+                case JsSuccess(s, x) =>
```
[the remark this answers is not in the document]
Comment (MazinGhafadgi) [reply app/controllers/AFTController.scala]: Done.

File: app/utils/JSONPayloadSchemaValidator.scala (hunk 1 above)
Comment (MazinGhafadgi) [reply app/utils/JSONPayloadSchemaValidator.scala]: Done

**Pull request merged**